### ISSUE TITLE
Trying to build wine-tkg on Opensuse Tumbleweed

### ISSUE THREAD

**Issue by Jordi234** [Tk-Glitch/PKGBUILDS#288]
Hi again!
On my hard try to find an alternative to Antergos, I'm trying a lot of distros. Now, I'm on Opensuse Tumbleweed, is really fast in comparation to Arch, I like it a lot.
Well, I see that now is possible to compile wine-tkg git for other distros, and I'm trying to do it.
I installed apparently all deps on my system, but the compilation fails here:

> gstreamer-1.0 base plugins 32-bit development files not found

I installed gstreamer-plugins-base-devel-32bit, and is the package that I can see near to the package I mentioned above, nothing more seems Opensuse has.
Where is this package on Opensuse?
Thanks!

**Comment by Tk-Glitch** [comment]
I haven't yet tried the script on OpenSuse Tumbleweed so I'm a virgin on that matter. I don't know how relevant this still is, but there seems to be some interesting notes about gstreamer devel deps here : https://wiki.winehq.org/OpenSUSE
Hopefully that'll get you there or at least closer.

Feel free to share your findings as the plan is to be able to add a page to the wiki with some helpful information to build on most distros. My testing so far has been limited to Ubuntu and Debian where I was able to build both wine-tkg and proton-tkg. Someone has shared his success with Fedora as well, but it was ultimately lacking details so I might give it a go myself to be able to write down what you need there. The more experiences we have, the better we can make it.

**Comment by Jordi234** [comment]
Well, I installed the packages mentioned on your link, and nothing changed. I downloaded and ¿installed? this package https://www.archlinux.org/packages/multilib/x86_64/lib32-gst-plugins-base/, nothing changed.
Probably the error comes previously. Here is the complete error, I don't understand nothing, I'm starting on compiling things:

> checking gst/gst.h usability... no
> checking gst/gst.h presence... yes
> configure: WARNING: gst/gst.h: present but cannot be compiled
> configure: WARNING: gst/gst.h:     check for missing prerequisite headers?
> configure: WARNING: gst/gst.h: see the Autoconf documentation
> configure: WARNING: gst/gst.h:     section "Present But Cannot Be Compiled"
> configure: WARNING: gst/gst.h: proceeding with the compiler's result
> configure: WARNING:     ## ------------------------------------ ##
> configure: WARNING:     ## Report this to wine-devel@winehq.org ##
> configure: WARNING:     ## ------------------------------------ ##
> checking for gst/gst.h... no
> configure: error: gstreamer-1.0 base plugins 32-bit development files not found, GStreamer won't be supported.
> This is an error since --with-gstreamer was requested.

**Comment by Jordi234** [comment]
Well, with skeevy's message, a part of the error dissapeared, now I'm here:

> checking for gst_pad_new in -lgstreamer-1.0... no
> configure: error: gstreamer-1.0 base plugins 32-bit development files not found, GStreamer won't be supported.
> This is an error since --with-gstreamer was requested.

**Comment by Tk-Glitch** [comment]
From the link I have posted earlier :

> After installing the above, in Leap 15 you will need to make the following symlinks in /usr/lib or ./configure will complain about not being able to find the libraries:
> ```
> ln -s libgstreamer-1.0.so.0 libgstreamer-1.0.so
> ln -s libgstbase-1.0.so.0 libgstbase-1.0.so
> ln -s libGL.so.1 libGL.so
> ```

No idea if this is still a thing. Better check `/usr/lib` for those first I guess.

**Comment by Jordi234** [comment]
I'm on Tumbleweed, my mind skipped that part. Now I'm rebuilding with symlinks....

**Comment by Jordi234** [comment]
Now, I'm lost. The compilation stops apparently without errors, but I don't have any package installed. I configured external path to lutris runners folder. Here is the last lines:

>  --sysroot=../.. -m32 -fno-PIC -fasynchronous-unwind-tables -shared \
>   ../../../wine-mirror-git/dlls/jscript/jscript.spec -mno-cygwin activex.o array.o bool.o compile.o \
>   date.o decode.o dispex.o engine.o enumerator.o error.o function.o global.o jscript.o \
>   jscript_main.o json.o jsregexp.o jsstr.o jsutils.o lex.o math.o number.o object.o regexp.o \
>   string.o vbarray.o cc_parser.tab.o parser.tab.o jscript.res jscript_classes_r.res jsglobal_t.res \
>   -loleaut32 -lole32 -luser32 -ladvapi32 -pipe -O2 -ftree-vectorize
> make[1]: se sale del directorio '/home/jordi/Descargas/PKGBUILDS/wine-tkg-git/src/wine-tkg-staging-fsync-32-build/dlls/jscript'
>  -> Removed BIG_UGLY_FROGMINER - Ribbit
>  -> Removed Proton-tkg token - Valve Ribbit
>  -> exit cleanup done

**Comment by Tk-Glitch** [comment]
Are `krb5-devel` and `krb5-devel-32bit` packages installed?
For the external thing that won't do anything currently with these types of builds. You'll get your build in a dedicated folder in the wine-tkg-git dir and you'll have to copy it where you want by hand.

**Comment by Jordi234** [comment]
Well, I read (Installing the wine-32bit-build-deps package will install most of the -32bit packages listed above.) , and I made the (stupid) mistake install only this package. Now I'm working with all the dependencies mentioned on the link of wine web.
When I installed all the 32 bit list, appeared A LOT of conflicts, and A LOT of options to choose for resolve the conflicts (I not understand nothing). I selected by intuition all, and all is installed now (I think). On the process Steam dissapeared from system, but I reinstalled it without more problems.

**Comment by Jordi234** [comment]
New error (this is frustrating :-S):
In file included from ../../../wine-mirror-git/dlls/uxtheme/gtk-button.c:25:

> ../../../wine-mirror-git/dlls/uxtheme/uxthemegtk.h:28:10: fatal error: gtk/gtk.h: No existe el fichero o el directorio
>    28 | #include <gtk/gtk.h>
>       |          ^~~~~~~~~~~
> compilation terminated.
> make[1]: *** [Makefile:239: gtk-button.o] Error 1

**Comment by Tk-Glitch** [comment]
I have found `gtk3-devel-32bit` for 32-bit but couldn't find a devel package for 64-bit. It might be part of the gtk3 package?

**Comment by Jordi234** [comment]
Seems is very difficult to install packages here. I retyped all the packages from wine web, and a lot are not installed. When you enter zypper install x x x, appears conflicts, is very important select the appropiate option, and not always is the number 1. 
Now, if I execute the command says "nothing to do", all is installed.
Restarting compilation.

**Comment by Jordi234** [comment]
Not build, still in the same error.
https://pkgs.org/download/gtk3-devel
Here is gtk3-devel 64 bit. 
I have installed both 64 and 32.

**Comment by Tk-Glitch** [comment]
`/usr/include/gtk-3.0/gtk/gtk.h` is in there so that seems to be the right one. pkg-config/pkgconf doesn't seem to do its job? Unless you don't have it installed already.

**Comment by Jordi234** [comment]
I have installed pkgconf-pkg-config.
I see mingw32-cross-pkg-config, mingw32-pkg-config, python2 and 3-pkgconfig and ruby2.6-rubygem-pkg-config.
Is one of them nedded?

**Comment by Tk-Glitch** [comment]
Nope

**Comment by Jordi234** [comment]
I retyped the dependences that skeevy said, and again a lot of them not installed.
I suppose that this kind pkgconfig(egl), are included on pkg-config package.
Retrying.

**Comment by Jordi234** [comment]
I don't understand the chaos installing packages on OpenSuse.

**Comment by Jordi234** [comment]
Continuing this. I made a /.configure on PKGBUILDS/wine-tkg-git/src/wine-mirror-git, here is the last part:

> configure: libhal 32-bit development files not found, no legacy dynamic device support.
> configure: libsane 32-bit development files not found, scanners won't be supported.
> configure: GTK3 32-bit development files not found, GTK themes won't be supported.
> configure: OSS sound system found but too old (OSSv4 needed), OSS won't be supported.
> configure: libvulkan and libMoltenVK 32-bit development files not found, Vulkan won't be supported.
> configure: vkd3d 32-bit development files not found (or too old), Direct3D 12 won't be supported.

I updated wind gmnome next repo gtk3-dev-32bit and nothing changed. I think this is the problem related with the error at build.

**Comment by Tk-Glitch** [comment]
libhal, libsane, OSSv4 and vkd3d can be considered fully optional. libMoltenVK is not interesting on anything but a mac os machine.
However GTK3 is likely required for staging and that libvulkan message might be actually misleading and targeted to mac os (I would ignore it for now).

For the GTK3 part, a workaround might be to simply drop the related staging patchset by using `_staging_userargs="-W uxtheme-GTK_Theming"` in your wine-tkg cfg. Hopefully this is all that's needed.

**Comment by Jordi234** [comment]
Ok, let's try again.
I found an interesting command here:
https://forums.opensuse.org/showthread.php/475586-Compiling-wine-from-source-help-please
> zypper si -d wine
Apparently installs all the needed build dependences.

**Comment by Jordi234** [comment]
THAT'S IT!! Build finished correctly.
Now, I will reinstall OpenSuse and try again only doing the correct things.
Later I write here the steps.

**Comment by Jordi234** [comment]
Well, that's all. System reinstalled and builded success wine-tkg git. I made the build just after installed the system and nvidia propietary drivers. This time not appeared "GTK3 32-bit development files not found", but still stopping on the same uxtheming point. The only problem that appeared now is an error on boot, that I can't understand, I will search what happen later:

> systemd[1]: /usr/lib/systemd/system/isdn.service:3: Failed to add dependency on hylafax-hfaxd, ignoring: Invalid argument

Here is the steps I made from start, and on order that are appeared problems. I don't know if "zypper si -d wine-staging" command is needed, or only the deps from wine web are sufficient, this command installs 160 packages, I think is excessive (or not).

> (https://www.unix.com/man-page/suse/1/ccache/)
> sudo zypper install ccache
> sudo cp /usr/bin/ccache /usr/local/bin/
> sudo zypper install gcc-c++
> sudo ln -s /usr/local/bin/ccache /usr/local/bin/gcc
> sudo ln -s /usr/local/bin/ccache /usr/local/bin/g++
> sudo ln -s /usr/local/bin/ccache /usr/local/bin/cc
> sudo zypper install wine
> added repo http://download.opensuse.org/repositories/Emulators:/Wine/openSUSE_Tumbleweed/
> sudo zypper si -d wine-staging
> sudo zypper install alsa-devel-32bit capi4linux-devel-32bit dbus-1-devel-32bit fontconfig-devel-32bit 
> freeglut-devel-32bit freetype2-devel-32bit giflib-devel-32bit glib2-devel-32bit glibc-devel-32bit glu-devel-32bit 
> krb5-devel-32bit libcom_err-devel-32bit libgnutls-devel-32bit libgphoto2-devel-32bit libgsm-devel-32bit 
> libjpeg8-devel-32bit liblcms2-devel-32bit libOSMesa-devel-32bit libpcap-devel-32bit libpng16-compat-devel-32bit 
> libpulse-devel-32bit libSDL2-devel-32bit libtiff-devel-32bit libv4l-devel-32bit libX11-devel-32bit libXcomposite-devel-32bit 
> libXcursor-devel-32bit libXext-devel-32bit libXfixes-devel-32bit libXi-devel-32bit libXinerama-devel-32bit 
> libXxf86vm-devel-32bit libxml2-devel-32bit libXrandr-devel-32bit libXrender-devel-32bit libxslt-devel-32bit 
> Mesa-libGL-devel-32bit mpg123-devel-32bit ncurses-devel-32bit openal-soft-devel-32bit openldap2-devel-32bit 
> libopenssl-devel-32bit unixODBC-devel-32bit xz-devel-32bit zlib-devel-32bit
> sudo zypper install git
> git clone https://github.com/Tk-Glitch/PKGBUILDS.git
> add _staging_userargs="-W uxtheme-GTK_Theming" to customization.cfg
> installed gcc-c++-32bit
> installed windows:mingw:win32 8.2.0 https://software.opensuse.org/package/mingw32-cross-gcc
> sudo zypper install opencl-headers
> sudo zypper install glib2-devel-32bit libpackagekit-glib2-devel-32bit glib2-tools-32bit
> sudo zypper install gstreamer-plugins-base-devel-32bit
> cd usr/lib
> sudo ln -s libgstreamer-1.0.so.0 libgstreamer-1.0.so
> sudo ln -s libgstbase-1.0.so.0 libgstbase-1.0.so
> sudo ln -s libGL.so.1 libGL.so
> sudo zypper install libudev-devel-32bit
> sudo zypper install FAudio-devel-32bit
> sudo zypper install cups-devel-32bit patch
> sudo zypper install schedtool
> ./non-makepkg-build.sh

**Comment by Jordi234** [comment]
One question. I reinstalled the system and builded a package with mistake, I not installed this:

> configure: libva 32-bit development files not found, GPU video acceleration won't be supported.

Now I solved this error. The question is, this error is not important?:

> configure: libvulkan and libMoltenVK 32-bit development files not found, Vulkan won't be supported.

Vulkan is not an important part of wine?

**Comment by Tk-Glitch** [comment]
Vkd3d/D3D12 is not in a usable state at this point in time so you can ignore it.

Vulkan is important for DXVK and D9VK usage though, if your hardware supports it. Since you seem to have the 64-bit lib installed already, you should only need libvulkan1 32-bit.

Good that we have a confirmation disabling that staging patchset fixes the GTK errors at least. That's an acceptable compromise until/if we can find a way to fix the dependency.

**Comment by skeevy420** [comment]
You're just missing some more dependencies.  Based on what you posted I'd wager it's libva-devel-32bit, libvulkan1-32bit, as well as there should also be a libvulkan1-SomethingNvidia from the Nvidia proprietary driver.  

libMoltenVK is for MacOS only so you don't need that.

vkd3d is provided from this project but only for Arch and similar users.  Personally, I haven't had any success using it with the limited amount of DX12 games I own so you're probably not missing much if anything for the time being.

libva-gl-devel-32bit and other similar libva-gl-* packages might also be needed if they're not already installed.

Lol, beat me by a minute.

**Comment by Jordi234** [comment]
Wait a moment!!!!
configure: libhal 32-bit development files not found, no legacy dynamic device support.
configure: libsane 32-bit development files not found, scanners won't be supported.
configure: OSS sound system found but too old (OSSv4 needed), OSS won't be supported.
configure: vkd3d 32-bit development files not found (or too old), Direct3D 12 won't be supported.

Disappeared. I found the package here:
https://download.opensuse.org/distribution/leap/15.0/repo/oss/
Exactly vulkan-devel-32bit. I don't know if is a good idea adding Leap repo to TumbleWeed....
I will try to rebuild.

**Comment by skeevy420** [comment]
I wouldn't add it.  That gave me nothing but hell when I did it in February when I last ran anything Suse.

**Comment by Jordi234** [comment]
Ok, I erased it. I have the package installed, that's all.

**Comment by Jordi234** [comment]
Build successful. How is possible to build wine on 17 minutes!! On arch takes hours...I don't understand nothing.
Well, I put here the steps that I made. I made this after installed the system and nvidia propietary packages:

> (https://www.unix.com/man-page/suse/1/ccache/)
> sudo zypper install ccache

> sudo cp /usr/bin/ccache /usr/local/bin/

> sudo zypper install gcc-c++

> sudo ln -s /usr/local/bin/ccache /usr/local/bin/gcc

> sudo ln -s /usr/local/bin/ccache /usr/local/bin/g++

> sudo ln -s /usr/local/bin/ccache /usr/local/bin/cc

> sudo zypper install wine

> added repo http://download.opensuse.org/repositories/Emulators:/Wine/openSUSE_Tumbleweed/

> sudo zypper si -d wine-staging

> sudo zypper install alsa-devel-32bit capi4linux-devel-32bit dbus-1-devel-32bit fontconfig-devel-32bit 

> freeglut-devel-32bit freetype2-devel-32bit giflib-devel-32bit glib2-devel-32bit glibc-devel-32bit glu-devel-32bit 

> krb5-devel-32bit libcom_err-devel-32bit libgnutls-devel-32bit libgphoto2-devel-32bit libgsm-devel-32bit 

> libjpeg8-devel-32bit liblcms2-devel-32bit libOSMesa-devel-32bit libpcap-devel-32bit libpng16-compat-devel-32bit 

> libpulse-devel-32bit libSDL2-devel-32bit libtiff-devel-32bit libv4l-devel-32bit libX11-devel-32bit libXcomposite-devel-32bit 

> libXcursor-devel-32bit libXext-devel-32bit libXfixes-devel-32bit libXi-devel-32bit libXinerama-devel-32bit 

> libXxf86vm-devel-32bit libxml2-devel-32bit libXrandr-devel-32bit libXrender-devel-32bit libxslt-devel-32bit 

> Mesa-libGL-devel-32bit mpg123-devel-32bit ncurses-devel-32bit openal-soft-devel-32bit openldap2-devel-32bit 

> libopenssl-devel-32bit unixODBC-devel-32bit xz-devel-32bit zlib-devel-32bit
> sudo zypper install git

> git clone https://github.com/Tk-Glitch/PKGBUILDS.git

> add _staging_userargs="-W uxtheme-GTK_Theming" to customization.cfg

> installed gcc-c++-32bit

> installed windows:mingw:win32 8.2.0 https://software.opensuse.org/package/mingw32-cross-gcc

> sudo zypper install opencl-headers

> sudo zypper install glib2-devel-32bit libpackagekit-glib2-devel-32bit glib2-tools-32bit

> sudo zypper install gstreamer-plugins-base-devel-32bit

> cd usr/lib

> sudo ln -s libgstreamer-1.0.so.0 libgstreamer-1.0.so

> sudo ln -s libgstbase-1.0.so.0 libgstbase-1.0.so

> sudo ln -s libGL.so.1 libGL.so

> sudo zypper install libudev-devel-32bit

> sudo zypper install FAudio-devel-32bit

> sudo zypper install cups-devel-32bit patch

> sudo zypper install schedtool

> sudo zypper install libva-devel-32bit

> added https://download.opensuse.org/distribution/leap/15.0/repo/oss/

> sudo zypper install vulkan-devel-32bit

> removed https://download.opensuse.org/distribution/leap/15.0/repo/oss/

> ./non-makepkg-build.sh

Now all is working good, except an error at boot:

> systemd[1]: /usr/lib/systemd/system/isdn.service:3: Failed to add dependency on hylafax-hfaxd, ignoring: Invalid argument 

I will search what happen with this error.
I don't know if this is the way to do this, but it worked. I test the sudo zypper si -d wine-staging command, only to try, probably the best is follow the instructions on wine web.

**Comment by Tk-Glitch** [comment]
Makepkg is set to use only a single thread to build by default. So yeah that's slow af, until you set makepkg to use all your threads.

**Comment by Jordi234** [comment]
I configured makepkg to use all threads, j2, j3, nproc and nproc+1, following your comments posted on this site.

**Comment by Tk-Glitch** [comment]
Oh yeah, I forgot. Anyway I'm building wine under 5 minutes on my machine, on Arch. So Arch isn't the issue here. However Manjaro is pretty bad and I think that's what you were using, right? I'd blame Manjaro tbh :frog:

**Comment by Jordi234** [comment]
The time test that I mentioned another day, was showing 35 seconds on arch. On Opensuse is showing 23 seconds, I think Opensuse has a supreme tweak on his kernel.

**Comment by skeevy420** [comment]
It only takes around 15 minutes on my 8yo Manjaro box.  Manjaro needs the same tweaks that Arch needs in that regard.  Not to mention that setting the compressor correctly in makepkg effects the time a lot since it's also single threaded by default.

Anyhoo, vulkan-devel-* is called vulkan-headers in Tumbleweed and the OSS packages aren't needed since you're likely using some combination of ALSA and PulseAudio.

**Comment by Jordi234** [comment]
I installed vulkan-headers and the error appeared. I think vulkan-headers are not 32 bit...

**Comment by skeevy420** [comment]
It's source code.  Architecture agnostic.

**Comment by Jordi234** [comment]
Oh, but not solved the issue with ./configure....I think still showing the Vulkan won't be supported message

**Comment by skeevy420** [comment]
I'm just mention this because that Vulkan package is a year old and is Vulkan 1.0 and most of these up-to-date projects expect recent 1.1 packages.

I was really considering going to Tumbleweed when this issue appeared.  Suse has some top notch system management tools.  I just had issues with ZFS repositories because I was on .8 from git and they only had .7.9 outside of some 3rd party repos that broke other third party repos and dammit I broke my system :crying_cat_face: That's not an issue anymore.

**Comment by Jordi234** [comment]
Yes, I'm on true. I uninstalled vulkan-devel-32bit, and installed vulkan-headers and again:
configure: libvulkan and libMoltenVK 32-bit development files not found, Vulkan won't be supported.

**Comment by Tk-Glitch** [comment]
You want `libvulkan1-32bit`, not `vulkan-devel-32bit`.

**Comment by skeevy420** [comment]
I posted that one above....

**Comment by Jordi234** [comment]
Now I'm trying to use the wne builded whith Lutris, and with wine installed on system, I have version mismach error, and without wine on system:
wine-tkg-staging-fsync-4.14.r18.g4aa460b8/bin/wine: error while loading shared libraries: libwine.so.1: wrong ELF class: ELFCLASS64

**Comment by skeevy420** [comment]
I dunno.  Possibly run "ldd" on the wine binary from Lutris and see if you're missing any libraries it expects.  That's where I'd start.

Tumbleweed just finished downloading.  I'm probably gonna disappear for a few hours to install and get it all set up; possibly until tomorrow depending on how busy I get today.

Once this all gets figured out an OBS Repository can be set up so all we'd have to do is just add that repository.

**Comment by Jordi234** [comment]
wine client error:0: version mismatch 590/589.
Your wine binary was not upgraded correctly,
or you have an older one somewhere in your PATH.
Or maybe the wrong wineserver is still running?

**Comment by Jordi234** [comment]
ldd command shows this:

> linux-gate.so.1 (0xf7fc8000)
> 	libwine.so.1 => /usr/lib/libwine.so.1 (0xf7da0000)
> 	libpthread.so.0 => /lib/libpthread.so.0 (0xf7d78000)
> 	libc.so.6 => /lib/libc.so.6 (0xf7b90000)
> 	libdl.so.2 => /lib/libdl.so.2 (0xf7b88000)
> 	/lib/ld-linux.so.2 (0xf7f98000)

I think is trying to use system libraries and are not the same version of wine-tkg.

**Comment by Tk-Glitch** [comment]
How are you launching wine-tkg?

**Comment by Jordi234** [comment]
From Lutris with custom runner, with "PKGBUILDS/wine-tkg-git/non-makepkg-builds/wine-tkg-staging-fsync-4.14.r18.g4aa460b8/bin/wine" path.
To try skevy answer I launched it from wine-tkg/bin folder, ldd ./../.../bin/wine /.../game.exe, I don't know how to launch ldd from Lutris.

**Comment by Tk-Glitch** [comment]
What happens if you run `"/home/jordi/.local/share/lutris/runners/wine/wine-tkg-staging-fsync-4.14.r18.g4aa460b8/bin/wine-tkg" "/mnt/24900711-3d8d-4042-bbe3-7762ddf28edb/.Playonlinux/wineprefix/TOCA/drive_c/Program Files/Codemasters/Race Driver 3/RD3.exe"` in a terminal? (make sure all instances of wine/wineserver are closed before doing so) ?

**Comment by Tk-Glitch** [comment]
Be aware it'll use the default WINEPREFIX path (`~/.wine`) so you might want to add `WINEPREFIX="/mnt/24900711-3d8d-4042-bbe3-7762ddf28edb/.Playonlinux/wineprefix/TOCA"` before the rest of the command.

**Comment by Jordi234** [comment]
>  WINEPREFIX="/mnt/24900711-3d8d-4042-bbe3-7762ddf28edb/.Playonlinux/wineprefix/TOCA" "/home/jordi/.local/share/lutris/runners/wine/wine-tkg-staging-fsync-4.14.r18.g4aa460b8/bin/wine-tkg" "/mnt/24900711-3d8d-4042-bbe3-7762ddf28edb/.Playonlinux/wineprefix/TOCA/drive_c/Program Files/Codemasters/Race Driver 3/RD3.exe"
>        .---.`               `.---.
>     `/syhhhyso-           -osyhhhys/`
>    .syNMdhNNhss/``.---.``/sshNNhdMNys.
>    +sdMh.`+MNsssssssssssssssNM+`.hMds+
>    :syNNdhNNhssssssssssssssshNNhdNNys:
>     /ssyhhhysssssssssssssssssyhhhyss/
>     .ossssssssssssssssssssssssssssso.
>    :sssssssssssssssssssssssssssssssss:
>   /sssssssssssssssssssssssssssssssssss/
>  :sssssssssssssoosssssssoosssssssssssss:
>  osssssssssssssoosssssssoossssssssssssso
>  osssssssssssyyyyhhhhhhhyyyyssssssssssso
>  /yyyyyyhhdmmmmNNNNNNNNNNNmmmmdhhyyyyyy/
>   smmmNNNNNNNNNNNNNNNNNNNNNNNNNNNNNmmms
>    /dNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNd/
>     `:sdNNNNNNNNNNNNNNNNNNNNNNNNNds:`
>        `-+shdNNNNNNNNNNNNNNNdhs+-`
>              `.-:///////:-.`
> 
> wine client error:0: version mismatch 590/589.
> Your wine binary was not upgraded correctly,
> or you have an older one somewhere in your PATH.
> Or maybe the wrong wineserver is still running?
> jordi@localhost:~> 

I turned on the pc, and I executed the command. I think is not any wine running....

**Comment by Tk-Glitch** [comment]
Yeah that's our *well known issue* by now. The lutris builds are working because they are compiled on a Debian-based distro (Ubuntu). For some reason wine built on Debian/buntu works everywhere, but when you build it on other distros it's hit or miss. Seems to be a miss for OpenSuse TW. I don't know what causes it since the source and flags used are the same (unless there's some hardcoded stuff on Debian, maybe?).
See https://github.com/Tk-Glitch/PKGBUILDS/issues/199 for reference

One thing that would be worth a try is to add `-W configure-Absolute_RPATH` to your _staging_userargs array.
So, that gives us `_staging_userargs="-W uxtheme-GTK_Theming -W configure-Absolute_RPATH"` with the GTK patchset.

If that doesn't help, that'll be the end of the road for now.

**Comment by Jordi234** [comment]
Sorry, I can't understand all the technical things...

> add -W configure-Absolute_RPATH to your _staging_userargs array.
Where is my _staging_userargs array? In which file?

> _staging_userargs="-W uxtheme-GTK_Theming -W configure-Absolute_RPATH" with the GTK patchset.

Is this a thing to remember? 

> _staging_userargs="-W uxtheme-GTK_Theming" in your wine-tkg cfg

**Comment by Tk-Glitch** [comment]
Yes it's in your cfg.
You have added `_staging_userargs="-W uxtheme-GTK_Theming"` to workaround the GTK issue (so we want to keep this).

So you should try to replace/update it with this `_staging_userargs="-W uxtheme-GTK_Theming -W configure-Absolute_RPATH"` then rebuild.

**Comment by Jordi234** [comment]
Rebuilding.
One thing, ony for curiosity I'm investigated some files on wine-tkg folder. On one of them, ine-tkg-staging-fsync-4.14.r18.g4aa460b8/bin/winepath, at the end of the file, is writed:
```
# finally, the default bin directory
if [ -x "/home/jordi/Descargas/PKGBUILDS/wine-tkg-git/wine-tkg-staging-fsync-4.14.r18.g4aa460b8/bin/wine" ]; then exec "/home/jordi/Descargas/PKGBUILDS/wine-tkg-git/wine-tkg-staging-fsync-4.14.r18.g4aa460b8/bin/wine" "$appname" "$@"; fi
if [ -x "/home/jordi/Descargas/PKGBUILDS/wine-tkg-git/wine-tkg-staging-fsync-4.14.r18.g4aa460b8/bin/wine64" ]; then exec "/home/jordi/Descargas/PKGBUILDS/wine-tkg-git/wine-tkg-staging-fsync-4.14.r18.g4aa460b8/bin/wine64" "$appname" "$@"; fi
```
I don't understand nothing but is pointing to "/home/jordi/Descargas/PKGBUILDS/wine-tkg-git/wine-tkg-staging-fsync-4.14.r18.g4aa460b8/bin/wine" but the folder is on "/home/jordi/Descargas/PKGBUILDS/wine-tkg-git/non-makepkg-builds/wine-tkg-staging-fsync-4.14.r18.g4aa460b8.
I don't know if this is a relevant thing.

**Comment by Tk-Glitch** [comment]
The winepath values are not a problem. You can check the lutris builds and you'll see that points to the buildbot path.

Ok.. sad. End of the road for OpenSuse TW for now, sadly. If anyone passing by has an idea, please share.

**Comment by Jordi234** [comment]
Well, it has been tried.
I have a good time trying and learning on the Linux world, is very interesting.
I will stay on Opensuse TW, is impressibely fast as I said, I like it. 
Not being able to use wine-tkg is not really important, works very well with the "official" builds that comes on Lutris, but I would like to try wine-tkg, of course.
Thanks for all !!

**Comment by Tk-Glitch** [comment]
The lutris builds are made with wine-tkg, you're only losing the ability to customize them to your liking. That being said, a Debian or Ubuntu VM could be used as a workaround. It has its fair share of dependencies to resolve but the resulting builds will work on any distro.
I'd really like to understand why we have that issue in the first place though. And why Debian-based doesn't care.

**Comment by Jordi234** [comment]
` you're only losing the ability to customize them to your liking`

But....I believed that building wine creates a "local" wine, optimized for my machine. I have thought  to use VM, but, using VM this "local" wine is working too??
On Antergos the performance change I noticed between lutris-wine and tkg-build, was spectacular.
On Opensuse TW, the performance change is system-general, and Lutris is not the exception.
Sorry I'm a little bit lost here xD

**Comment by skeevy420** [comment]
I actually got Tumbleweed installed then I broke its Grub testing a community Grub 2.04 and full disk BTRFS Zstd compression...just gonna use lzo on the next install (it does work).

Then I went to update a few things from here on Manjaro before bed, forgot to manually edit llvm-git to use the release/9.x branch so I built 10 which does not like mesa-aco right now so I'm rebuilding all of that at the moment.

I have been documenting my entire Tumbleweed install, programs/patterns manually added, and all the changes I've made to narrow this down when I get back to that point.

**Comment by skeevy420** [comment]
@Jordi234 It's both.  When you enable the local_optimized setting and set the flags right, it does produce optimized binaries tailor made for your specific hardware...or disable that setting and use your makepkg.conf settings...

There are also the options for enabling various fixes & tweaks like the KDE system tray fix or Fsync for regular Wine.  You lose the ability to pick and choose what fixes and tweaks you need or don't need when going with the Lutris builds.

**Comment by Jordi234** [comment]
I'm using LXDE. I'm interested with this practically only for the local_optimized option. I have few (old) games running very slow, and I can't determine if is the slow nvidia or playing trough wine.

**Comment by Jordi234** [comment]
I go to try to install lubuntu on a usb stick and make the build from there :-P

**Comment by skeevy420** [comment]
Quick question -- Did you enable the Sources repository in Suse?

**Comment by Jordi234** [comment]
No, is not active. Why? Is needed?

**Comment by skeevy420** [comment]
Just to be able to build a package locally easier.  The way I figure, if I can't build Wine the way Tumbleweed expects then I probably can't build it in a customized manner.

Though, looking into it more, with how Suse wants to do things using their OBS tools, the rpmbuild and source package method doesn't seem to be very supported.  Looks like I'm going to have to sign up to that service figure out that method of building packages.

But I am in the process of using rpmbuild to build Tumbleweed's wine and so far the compile hasn't errored out..just waiting on it to finish the find-* stuff near the end of the build (which is taking like 2-3x longer than the build), and if successful, I'm gonna move on to building my usual wine setup from here with the script.

**Comment by Jordi234** [comment]
I opened the script and I found:

```
 ./non-makepkg-build.sh --deps64
./non-makepkg-build.sh --deps32
```
This commands not work on Opensuse? Is this the way to know if has all the dependences?
I will try it later, now I'm building on Anarchy installed on usb stick :)

**Comment by skeevy420** [comment]
Use "sudo zypper in -d *name_of_package*" to get and install Tumbleweed's expected build dependencies.

**Comment by skeevy420** [comment]
Just to update, have Wine64 w/ vkd3d building on Tumbleweed and working on figuring out Wine32 build dependencies.  No workarounds necessary.

**Comment by skeevy420** [comment]
Suse is a nightmare...missing lots of 32bit dependencies, wants us to sign up for a build service, compiler says you're missing stuff you know you have....I know damn well OpenGL is installed....

I don't see how people put up with it.  It's a nice OS right up until you want something it doesn't provide.

**Comment by Jordi234** [comment]
Is not much appreciated, appears on 9th place on distrowatch for last 12 months HPD 809, MXLinux is 1st with HPD 3906, but I repeat, it's the fastest distro I've tried on my pc by far, I thought it was Arch, but no.

**Comment by Jordi234** [comment]
I don't give up, there are still things to try. At first steps that I made, I used a strange zypper command to install dependences. Now I followed the instructions on wine web, I installed patterns-openSUSE-devel_basis, bison-32bit, and all the first list. The result, more than 70 packages installed. Rebuilding

**Comment by Jordi234** [comment]
A little offtopic (erase it later).
Is possible to build a "local_optimized" wine from virtualbox? A wine optimized for my machine from virtualbox.

**Comment by Tk-Glitch** [comment]
Using a VM to build is probably the best you can do when your distro is too problematic to build on. I'm slowly working on a potential solution for such cases looking forward but I feel there will always be a corner case that frogs us over. We'll see how it goes I guess.

**Comment by Jordi234** [comment]
I used anarchy installed on a USB pen, it is faster and more convenient than a VM. It is a shame not to be able to use Opensuse for this task because it is the fastest distro I have ever seen, much more than Arch. The kernel or configuration it uses makes everything fly, seriously. I'm still in Tumbleweed.

**Comment by Jordi234** [comment]
Hi again friends!
Today I tried another "way" to do this build. A few days ago, I found (searching I don't remember what), a forum talking about building wine on OpenSuse, and I saw an answer recommending this file:
https://build.opensuse.org/package/view_file/home:junknot/wine/wine.spec?expand=1
Well, I made a list from this file, with the deps that I saw, and mounted a VBox with Tumbleweed. After a little little installation around 600 packages, and installing few other packages, the build started very well (I not made the config to see the missing deps exactly), and, when 32 bit build started, I had the same problems again, identical from this question that I asked here.
I don't understand nothing, is impossible to build this wine version on OpenSuse TW?? The Gtk3 32bit dev missing libraries error is surrealist...

**Comment by Tk-Glitch** [comment]
Sadly multilib environments aren't supported equally accross distros and it can be very frustrating to work with. I might give it look myself by installing TW since I will have to install more broken distros to make wiki documentation in the coming days. Hopefully I'll find a fix I can share with you and other TW users.

**Comment by Jordi234** [comment]
A little detail, I don't know if is useful. I'm trying to know differences between Arch and OpenSuse, for the gtk3, and I see a little detail. Arch have a folder /usr/lib32/pkgconfig/, and this folder have a lot of files. This folder but on OST, may be /usr/lib/pkgconfig, but only have glib-2.0.pc file. Viewing the files on lib32-gtk3 package from arch (https://www.archlinux.org/packages/multilib/x86_64/lib32-gtk3/), I can see many things identical, for example, the things under /usr/lib32/, are the same on OST on /usr/lib, but on pkgconfig, OST is "empty". This detail can make fail the configure when is starting build?

**Comment by mrxz** [comment]
I managed to build it on both openSUSE Leap 15.2 and Tumbleweed without too much hassle. It's basically a question of installing the missing -devel(-32bit) packages. Only on Tumbleweed did I also run into the _"gstreamer-1.0 base plugins 32-bit development files not found"_ problem, despite `gstreamer-plugins-base-devel-32bit` being installed. This error is misleading, as the actual problem is that `glib2-devel-32bit` is missing (which is needed by gstreamer). In hindsight the same is probably true for 15.2, but I must've had it already installed.

After installing it, wine does compile. However, the script would build the 64-bit version twice. To combat this, I had to make the following change in `build.sh`:
```diff
	  # /nomakepkg
	  if [[ ! ${_makepkg_options[*]} =~ "ccache" ]] && [ -e /usr/bin/ccache ]; then
-	    export CC="ccache gcc"
+	    export CC="ccache gcc -m32"
-	    export CXX="ccache g++"
+	    export CXX="ccache g++ -m32"
	  fi
```
Not sure if this is supposed to be inferred in some other way that doesn't work on openSUSE, but at least this in line with what is documented on the WineHQ Wiki ([OpenSUSE Building 32-bit Wine](https://wiki.winehq.org/OpenSUSE#Building_32_bit_Wine)).

There was one more oddity that I only observed on 15.2. I had to set `_pkg_strip` to false, as somehow the stripping broke the compiled wine. On Tumbleweed this wasn't needed. (Although I can't rule out that this is somehow due to my 15.2 setup).

In short; all I had to do was install the right development packages, make the above tweak to `build.sh` and (optionally) disable stripping of libraries.

**Comment by cglmrfreeman** [comment]
Managed to build `wine-tkg-staging-fsync-git-6.4.r2.gc747e46d` on TW after resolving all dependency issues. Lutris sees it and it's a selectable Wine runner, however when attempting to launch the Battle Net launcher, I receive the following error: `wine: could not load ntdll.so: /home/rob/.local/share/lutris/runners/wine/wine-tkg-staging-fsync-git-6.4.r2.gc747e46d/bin/../lib64/wine/ntdll.so: wrong ELF class: ELFCLASS64`

Was this ever going to be looked into any further? I'd be more than happy to help, as lutris-6.0 crashes my system and gets me bans in the game.

**Comment by Jordi234** [comment]
This is a BIG offtopic, but I tried proton-ge-custom and it worked with many games that wine can't, and a little bit more faster than wine.
Try it if you like.
Now I'm on Debian Lxde, TW is crazy with Nvidia drivers normally.
